ios/RNGoogleSignin/RNGoogleSignin.m

```diff
@@ -97,24 +161,24 @@ - (void)signIn:(GIDSignIn *)signIn didSignInForUser:(GIDGoogleUser *)user withEr
                            @"accessTokenExpirationDate": [NSNumber numberWithDouble:user.authentication.accessTokenExpirationDate.timeIntervalSinceNow]
                            };
 
-    return [self.bridge.eventDispatcher sendAppEventWithName:@"RNGoogleSignInSuccess" body:body];
-}
+    resolve(body);
 
-- (void) signInWillDispatch:(GIDSignIn *)signIn error:(NSError *)error {
-    return [self.bridge.eventDispatcher sendAppEventWithName:@"RNGoogleSignInWillDispatch"
-                                                        body:@{}];
+    self.currentUserAsyncResolve = nil;
+    self.currentUserAsyncReject = nil;
+    self.signInResolve = nil;
+    self.signInReject = nil;
+    self.revokeAccessResolve = nil;
+    self.revokeAccessReject = nil;
+    self.isSigningInSilently = NO;
+    return;
 }
 
 - (void)signIn:(GIDSignIn *)signIn didDisconnectWithUser:(GIDGoogleUser *)user withError:(NSError *)error {
   if (error != Nil) {
-    return [self.bridge.eventDispatcher sendAppEventWithName:@"RNGoogleRevokeError"
-                                                        body:@{
-                                                               @"message": error.description,
-                                                               @"code": [NSNumber numberWithInteger: error.code]
-                                                               }];
+    return RNGoogleSigninRejectWithError(@"Error when revoking access", error, self.revokeAccessReject);
   }
 
-  return [self.bridge.eventDispatcher sendAppEventWithName:@"RNGoogleRevokeSuccess" body:@{}];
+  return self.revokeAccessResolve(nil);
```

Comment:
I don't know the exact flow and possible outcomes here, but you set `revokeAccessResolve` to nil on line 170. If you try to execute a block that is nil, your program will crash. It is good practice to check this before calling the block, like:
```
if (self.revokeAccessResolve() {
  self.revokeAccessResolve(nil)
}
```

Reply:
It should not be possible to call `self.revokeAccessResolve` (or other resolves/rejects). And if it is something has gone wrong and I think crash is appropriate in those situations since nothing would work anyway and we cannot recover from that.

I probably should have a check that in the beginning of `signIn` method (line 70) revoke/reject methods are not `nil`.